maxima/g0/deflate_geo/deflate-geo.mac

```diff
@@ -0,0 +1,72 @@
+load("modal-basis")$
+load(stringproc)$
+load("out-scripts")$
+
+/* Taking DG fields g_ij and |B| as inputs, this kernel will calculate the other geometric quantities
+J = det(g_ij))
+b_i = g_iz/sqrt(g_zz)
+*/
+
+deflateGeo(fh, basisName, polyOrder) := block(
+/* powerset maxima function will give me combos*/
+
+  disp("basisName", basisName),
+  /* get the full 3d basis*/
+  kill(varsC, varsP, basisC, basisP),
+  [vars,basis] : loadBasis(basisName, 3, polyOrder),
+  numCoeffs : length(basis),
+
+  /* We can never rmove z*/
+  varsNoZ : delete(z,vars),
+  /*kill(varsC, varsP, basisC, basisP),*/
+  /*[varsNoZ,basisNoz] : loadBasis(basisName, ndim, polyOrder),*/
+  removeVarsList : delete([],full_listify(powerset(setify(varsNoZ)))),
+
+  disp("list of remove vars", removeVarsList),
+
+  for i : 1 thru length(removeVarsList) do(
+    removeVars : removeVarsList[i],
+    /*[vars_deflated,basis_deflated] : loadBasis(basisName, ndim, polyOrder),*/
+    vars_deflated : vars,
+    removeStr : "rem",
+    for j : 1 thru length(removeVars) do( 
+      vars_deflated : delete(removeVars[j], vars_deflated),
+      removeStr : sconcat(removeStr,removeVars[j])
+    ),
+    ndim : length(vars_deflated),
+    disp("vars_deflated: ", vars_deflated),
+    disp("removeVars: ", removeVars),
+    disp("Vars", vars),
```

Comment:
if this is helpful when the script is running, keep it. If it was for dev, delete it

Reply:
this applies to all the functions in this PR